blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx

```diff
@@ -119,12 +119,13 @@ class RunDetailsHeader extends Component {
             </div>
         );
 
-        const commitIdString = run.commitId || '—';
+        const commitIdString = changeSet.length && changeSet[changeSet.length - 1].commitId;
```

Comment:
I removed the '-' attribution because it is indeed handled by the CommitId component, we do still need to check for changeSet.length to not go out of bounds. (I tried to do it intentionally to see what would happen and mobx threw an out of bounds error)